contracts/ControllerEllipsisLPStrat.sol

```diff
@@ -0,0 +1,219 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.11;
+
+import "./ControllerStratAbs.sol";
+import "../interfaces/IEps.sol";
+import "../interfaces/IWNative.sol";
+
+contract ControllerEllipsisLPStrat is ControllerStratAbs {
+    using SafeERC20 for IERC20;
+    using SafeERC20 for IERC20Metadata;
+
+    address constant public REWARD_TOKEN = address(0xA7f552078dcC247C2684336020c03648500C6d9F);
+    address immutable public POOL_TOKEN; // 0x5781041F9Cf18484533F433Cb2Ea9ad42e117B3a BNB
+    IEpsLPPool immutable public POOL; // 0xc377e2648E5adD3F1CB51a8B77dBEb63Bd52c874 BNB
+    IEpsStaker constant public STAKE = IEpsStaker(0xcce949De564fE60e7f96C85e55177F8B9E4CF61b);
+    IEpsMultiFeeDistribution constant public FEE_DISTRIBUTION = IEpsMultiFeeDistribution(0x4076CC26EFeE47825917D0feC3A79d0bB9a6bB5c);
+
+    int128 private immutable TOKEN_INDEX; // want token index in the pool
+    uint private constant TOKENS_COUNT = 2; // LP pool
+    uint private immutable STAKE_POOL_ID; // 11 BNB/BNBL (bnbEPS)
+
+    constructor(
+        IERC20Metadata _want,
+        uint _stakePoolId,
+        int128 _tokenIndex,
+        address _poolToken,
+        address _pool,
+        address _controller,
+        address _exchange,
+        address _treasury
+    ) ControllerStratAbs(_want, _controller, _exchange, _treasury) {
+        POOL = IEpsLPPool(_pool);
+        POOL_TOKEN = _poolToken;
+        STAKE_POOL_ID = _stakePoolId;
+        TOKEN_INDEX = _tokenIndex;
+    }
+
+    // Deposit Native
+    receive() external payable { }
+
+    function identifier() external view returns (string memory) {
+        return string(abi.encodePacked(
+            IERC20Metadata(POOL_TOKEN).symbol(), "@Ellipsis#1.0.0"
+        ));
+    }
+
+    function harvest() public nonReentrant override {
+        uint _before = wantBalance();
+
+        _claimRewards();
+        _swapRewards();
+
+        uint harvested = wantBalance() - _before;
+
+        // Charge performance fee for earned want + rewards
+        _beforeMovement();
+
+        // re-deposit
+        if (!paused()) { _deposit(); }
+
+        // Update lastBalance for the next movement
+        _afterMovement();
+
+        emit Harvested(address(want), harvested);
+    }
+
+    function _deposit() internal override {
+        uint wantBal = wantBalance();
+
+        if (wantBal > 0) {
+            uint[TOKENS_COUNT] memory amounts = _amountToAmountsList(wantBal);
+
+            uint expected = _wantToPoolTokenDoubleCheck(wantBal, true);
+
+            IWNative(address(want)).withdraw(wantBal);
```

Comment:
this line should be inside an `if (address(want) == WNATIVE) ` same than `deposit{}` because for the other strategies will not work (or keep it like this and rename the SC to `EllipsisNativeLP` or something like that =P)

Reply:
Great, you are right, far better with the `if native`, at least for now =)